### DISCUSSION TITLE
Error cloning crazyflie-firmware repo

### DISCUSSION THREAD

**Discussion by hanqiuli** [bitcraze/discussions#967] [Q&A] [answered]
Hi, when cloning the firmware repository I get the following error

fatal: Unable to checkout 'a65b7c9a3e6502127fdb80eb288d8cbdf251a6f4' in submodule path 'vendor/CMSIS'

This gives me problems when I try to build the firmware of the crazyflie.
I've tried using git submodule init and update, and I get the same error. 

I've seen on a previous post that this might be a problem with git. Is there any workaround?

**Comment by knmcguire** [comment, marked as answer]
Assuming you did the following comand to get both the submodules and the main repo:

```
git clone git@github.com:bitcraze/crazyflie-firmware.git --recursive
```

I get the following:

```
Resolving deltas: 100% (4100/4100), done.
Submodule path 'vendor/CMSIS': checked out 'a65b7c9a3e6502127fdb80eb288d8cbdf251a6f4'
Submodule path 'vendor/FreeRTOS': checked out '3604527e3b31991c596cd420f32989ee890aca4a'
```

It seems to be working for me... maybe there was some kind of firewall or network error that was blocking this? The commit hash does seem to exist. 

Try to remove the repo and do a clean clone with --recursive again?
